src/ensenso.cpp

```
@@ -969,4 +969,40 @@ Result<Ensenso::CaptureParams> Ensenso::getCaptureParameters(bool crop_to_roi) {
 	return params;
 }
 
+bool Ensenso::isDumpingTree() {
+	return dump_tree_;
+}
+
+void Ensenso::enableNxLibLogging(std::string connection_folder_path, std::string debug_level, int log_file_size, bool dump_tree) {
+	NxLibItem debugOut = root[itmDebug][itmFileOutput];
+	debugOut[itmFolderPath] = connection_folder_path;
+	debugOut[itmMaxTotalSize] = log_file_size;
+	debugOut[itmEnabled] = true;
+	root[itmDebug][itmLevel] = debug_level;
+	dump_tree_ = dump_tree;
+	connection_folder_path_ = connection_folder_path;
+}
+
+void Ensenso::dumpTree(std::string time_stamp) {
+	if (connection_folder_path_.empty()) {
+		log("failed to dump camera tree. NxLibLogging is not enabled.");
+		return;
+	}
+
+	Result<std::string> serial_number = serialNumber();
+	if (!serial_number) log("failed to determine stereo camera serial number");
```

```suggestion
	if (!serial_number) log("Failed to determine stereo camera serial number.");
```

---

Also we can't continue if there's no serial. We should return an error then I guess? At the moment, if there's no serial, it will log it and then crash :).